### PR TITLE
feat: implement JUMPI instruction

### PR DESCRIPTION
**Pull request by lambda-0x**
## Pull Request type

Please check the type of change your PR introduces:

- [x] Feature

## What is the current behavior?

Resolves: #252 

## What is the new behavior?

- implement `JUMPI` opcode
- merge after #355 

## Does this introduce a breaking change?

- [x] No

### CODE REVIEW

File: crates/evm/src/instructions/memory_operations.cairo [hunk 1]
```diff
@@ -86,6 +86,19 @@ impl MemoryOperation of MemoryOperationTrait {
...
     /// # Specification: https://www.evm.codes/#57?fork=shanghai
     fn exec_jumpi(ref self: ExecutionContext) -> Result<(), EVMError> {
+        // Peek the value so we don't need to push it back again incase we want to call `exec_jump`
+        let b = self.stack.peek_at(1)?;
+
+        if b != 0x0 {
+            self.exec_jump()?;
+            // counter would have been already popped by `exec_jump`
+            // so we just remove `b`
+            let _ = self.stack.pop()?;
```
Comment (Eikix) [inline crates/evm/src/instructions/memory_operations.cairo:96]:
```suggestion
            self.stack.pop()?;
```

File: crates/evm/src/instructions/memory_operations.cairo [hunk 2]
```diff
@@ -86,6 +86,19 @@ impl MemoryOperation of MemoryOperationTrait {
...
+        } else {
+            // remove both `value` and `b`
+            let _ = self.stack.pop_n(2)?;
```
Comment (Eikix) [inline crates/evm/src/instructions/memory_operations.cairo:99]:
```suggestion
            self.stack.pop_n(2)?;
```

**Review by Eikix — COMMENTED** [review]
can you rebase as well to have "true diff" on tests:)?

**Comment by lambda-0x** [comment]
done

**Comment by Eikix** [comment]
Just remembered there is a supported_opcodes.md file in the documentation, can you update with JUMPI, JUMP and recently implemented opcode 😇?

**Comment by lambda-0x** [comment]
fixed both, i went through the commit history and have updated OPCODEs that i saw there, LMK if there are any that i missed.

**Review by Eikix — COMMENTED** [review]
SWAP1 to 16 and lgtm 🔥

**Review by enitrat — DISMISSED** [review]

**Review by Eikix — APPROVED** [review]
lgtm 🔥

**Pull request merged**